from Schedule import *
import io
import base64
import yaml
import matplotlib
matplotlib.use('Agg') # используем Agg - неинтерактивный бэкэнд, который записывает графики в файлы ( в нашем случае в последовательность байтов), без использования дисплея


# Минималистичный формат
yamlMin = yaml.safe_load("""
    x1Max: 95
    x2Max: 95
    lines:
    - "x1"                    
    - "1 + log2(x1)"
    points:
    - [[2, 2],[3, 3],[4, 4],[5, 5],[6, 6],[7, 7],[8, 8],[9, 9],[10, 10],[15, 15],[20, 20],[25, 25],[30, 30],    [35, 35],[40, 40],[45, 45],[50, 50],[100, 100],[150, 150],[200, 200],[250, 250],[300, 300],[350, 350],    [400, 400],[450, 450],[500, 500],[1000, 1000],[2000, 2000],[3000, 3000],[3500, 3500],[3510, 3510],    [3550, 3550],[3600, 3600],[3700, 3700],[3800, 3800],[3900, 3900]]

    - [[2, 2],[3, 3],[4, 3],[5, 4],[6, 4],[7, 4],[8, 4],[9, 5],[600, 11],[15, 5],[20, 6],[25, 6],[30, 6],[35, 7],    [40, 7],[45, 7],[50, 7],[100, 8],[150, 9],[200, 9],[250, 9],[300, 10],[350, 10],[400, 10],[450, 10],[500, 10],    [700, 11],[800, 11],[900, 11],[1000, 11],[1100, 12],[1200, 12],[1300, 12],[1400, 12],[1500, 12],[1600, 12],    [1700, 12],[1800, 12],[1900, 12],[2000, 12],[2100, 13],[2200, 13],[2300, 13],[2400, 13],[2500, 13],[2600, 13],    [2700, 13],[2800, 13],[2900, 13],[3000, 13],[3100, 13],[3200, 13],[3300, 13],[3400, 13],[3500, 13],[3600, 13],    [3700, 13],[3800, 13],[3900, 13],[4000, 13],[4100, 14],[4200, 14],[4300, 14],[4400, 14],[4500, 14],[5000, 14],    [6000, 14],[7000, 14],[8000, 14],[9000, 15],[10000, 15],[11000, 15],[12000, 15],[13000, 15],[14000, 15],[15000, 15],    [20000, 16],[30000, 16]]
""")




from PIL import Image

if __name__ == '__main__':
    try:
        Schedule.MakePlot(yamlMin)
        base64Encode = Schedule.PlotToBase64()
        
        # decode image and display that
        base64Decode = base64.b64decode(base64Encode);
        bytesIO = io.BytesIO(base64Decode);
        img = Image.open(bytesIO)   
        img.show()

    except Schedule.Exception as err:
        print(err)